Design/ThirdpartyDependencies.md

```diff
@@ -0,0 +1,335 @@
+to ensure packaging doesn't interfere with system versions
+but this is something supported by Python
+and done by other systems like `Conda`.
+Python requirements could be easily supported by a requirements file
+or some newer system like
+[pipenv](https://packaging.python.org/key_projects/#pipenv) to properly
+define a frozen set of dependencies.
+Some packages such as `PyQt` do not provide the headers and link
+libraries required for the C++ build.
+The easiest solution would be a Conan package for `PyQt` much like macOS
+Homebrew does. 
+
+The availability of `mantid` to other Python libraries would be the cause
+of the most concern if we adopted the Conan approach as it is
+about creating a self-contained package.
+
+
+### Conda
+
+*"A conda package is a binary tarball containing
+system-level libraries,
+Python modules,
+executable programs,
+or other components"*.
+
+[Conda](https://docs.conda.io/en/latest/)
+is another cross-platform package manager but with support for a
+variety of languages including Python and C++.
+Alongside `pip` many popular Python packages support distribution through Conda.
+In fact `mantid` already has [Conda packages](https://anaconda.org/mantid/repo)
+for workbench and the framework for Linux.
+Mac packages are currently effectively abandoned.
+
+[Anaconda](https://repo.continuum.io/pkgs/) hosts many prebuilt packages for
+various combinations of compilers and operating systems.
+[Conda build](https://conda.io/projects/conda-build/en/latest/resources/commands/conda-build.html)
+is provided to enable building custom packages.
+Compilers are somewhat limited., getting latest versions of compilers on all platforms is
+problematic/hacky at best.
+
+The [meta.yaml](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#requirements-section)
+file provides a central place to define dependencies and would satisfy requirements 2-4.
+The requirements file allows builds tools such as CMake to be used.
+
+It is unclear how the build process would fit into current tools such as IDEs.
+The current CLI workflow is documented [here](https://github.com/mantidproject/conda-recipes/blob/master/Developer.md).
+
+Conda would possibly a better solution for providing just the `mantid` framework
+as a library for other users to import and not considering this as a solution
+for providing the full desktop application bundle. We also already have a
+working conda build of the framework so this is low overhead. This needs
+proper support though.
+
+## Summary + Questions
+
+After consideration I would suggest be that we adopt both options for supporting
+the two use cases presented. Both cases have different needs so there is no
+reason to expect one system to support both.
+
+Questions: 
+
+1. Do we need to still support a build against system libraries on Linux?
+
+   * Would SNS need an RPM framework build for their GUIs linked to system Python?
```

Comment:
ORNL would prefer to connect to a `mantid-qa` package that had the extra bits of qt from mantid that we use. We would need to take inventory of what objects/subpackages we use first.

Reply:
Could we do a `mantid-deps` and `mantid-extra-ornl` / `mantid-extra-*facility*` ...etc. similar to how the Linux kernel was packaged with Debian based systems

This way if your packaging Mantid elsewhere, it's obvious which meta packages you need. If there is enough packages, you could also have a `mantid-all` which includes all nested ones for CI jobs that can build anything?

Reply:
@peterfpeterson Would you want the `mantid-qt` package as an rpm built against system Python? My first thought for library-based usage was conda and have 2 conda packages:
* `mantid-framework`
* `mantid-qt`

and we would not have a framework/mantidqt rpm at all.

The rpm/deb would only be used for the workbench application for ease of distribution and installation on facility systems even though that deb/rpm would contain all of the required libraries and be self-contained.